pkg/cassandra/version/version.go

```diff
@@ -37,10 +37,28 @@ func New(s string) *Version {
 	return v
 }
 
+func (v *Version) set(s string) error {
+	sv, err := semver.NewVersion(s)
+	if err != nil {
+		return err
+	}
+	v.versionString = s
```

Comment:
Do we need to keep the versionString in memory? Surely `semver.String` will be more up to date?

Reply:
We need it because we want to remember the format of the originally supplied version string.
So if the Cassandra node reports version `3.11` via nodetool and a JMX query, we want to report that exact version in the `Pilot.Status.Cassandra.Version` rather than the semver `3.11.0`.

We also don't want a user to submit `CassandraCluster.Spec.Version: 3.11` and then have that change to `3.11.0` when they do `kubectl get CassandraCluster -o yaml`.

However, when we infer the Docker image tag, we use the semver, because we don't want docker to simply pull the latest `3.11` image from Dockerhub.